daffodil-io/src/main/scala/org/apache/daffodil/io/ByteBufferDataInputStream.scala

```diff
@@ -1,1270 +0,0 @@
-/*
```

Comment:
This file is removed in this diff. It's all removals.

Regarding the two implements, there are two implementations of InputSource's in InputSource.scala. There's the BucketingInputSource which implements the bucketing algorithm, and a ByteBufferInputSource which just reads data out of a ByteBuffer. There's no bucketing there, so it should be more efficient if someone already has data in an array or byte buffer, The downside is that we cannot free up data from a ByteBuffer so it might waste memory. That's also limited to 2GB because of Java. The BucketingInputSource has a much higher limit of MaxInt * bucketSize.